<launch>
    <arg name="neural_network_inference_type"/> <!-- Options: cpu, torch_gpu or trt_gpu -->

    <node pkg="robot_name_detector" type="robot_name_detector_node.py" name="robot_name_detector_node">
        <param name="fast_sound_rms_attack" value="0.05"/>
        <param name="fast_sound_rms_release" value="0.95"/>
        <param name="slow_sound_rms_attack" value="0.975"/>
        <param name="slow_sound_rms_release" value="0.975"/>
        <param name="sound_presence_relative_threshold" value="2.5"/>

        <param name="inference_type" value="cpu"/>
        <param name="robot_name_model_probability_threshold" value="0.05"/>
        <param name="robot_name_model_interval" value="1600"/>
        <param name="robot_name_model_analysis_delay" value="5600"/>
        <param name="robot_name_model_analysis_count" value="3"/>

        <param name="led_status_duration_s" value="1.0"/>

        <remap from="audio_in" to="audio_input_signed_16_16000"/>
        <remap from="audio_in/filter_state" to="robot_name_detector/filter_state"/>
        <remap from="led_status/filter_state" to="robot_name_detector/led_status/filter_state"/>
    </node>

</launch>
